src/mui/form/SimpleForm.js

```diff
@@ -6,18 +6,22 @@ import getDefaultValues from './getDefaultValues';
 import FormField from './FormField';
 import Toolbar from './Toolbar';
 
-export const SimpleForm = ({ children, handleSubmit, invalid, record, resource, basePath }) => (
-    <form onSubmit={handleSubmit} className="simple-form">
-        <div style={{ padding: '0 1em 1em 1em' }}>
-            {React.Children.map(children, input => input && (
-                <div key={input.props.source} style={input.props.style}>
-                    <FormField input={input} resource={resource} record={record} basePath={basePath} />
-                </div>
-            ))}
-        </div>
-        <Toolbar invalid={invalid} />
-    </form>
-);
+export const SimpleForm = ({ children, handleSubmit, invalid, record, resource, basePath, submitOnEnter }) => {
+    const formOnSubmit = submitOnEnter ? handleSubmit : function noop() {};
```

Comment:
this will create a new `noop()` function for every render, and therefore force a rerender of children even if nothing changed. You should move `noop()` definition out of the component.

Also, no need for a `const`, you can do that directly in the `onSubmit` prop, in JSX.